packages/ts-plugin/src/index.ts

```diff
@@ -4,27 +4,41 @@ import type { CollectionConfig } from './frontmatter.js';
 import { getFrontmatterLanguagePlugin } from './frontmatter.js';
 import { getLanguagePlugin } from './language.js';
 
-export = createLanguageServicePlugin((ts, info) => {
-	let collectionConfig = undefined;
-
+function getCollectionConfig(
+	readFile: (path: string) => string | undefined,
+): CollectionConfig['config'] | undefined {
 	try {
-		const currentDir = info.project.getCurrentDirectory();
-		const fileContent = ts.sys.readFile(currentDir + '/.astro/collections/collections.json');
+		let fileContent = readFile('/.astro/astro/collections/collections.json');
+		if (fileContent) {
+			return JSON.parse(fileContent);
+		}
+		// If we couldn't find a config there, the project might be an Astro 4 project
+		fileContent = readFile('/.astro/collections/collections.json');
```

Comment:
Here the logic could be simpler, you can check if `fileContent`, if not, redefine it, and only after all that use `JSON.parse` instead of having the if twice and two JSON.parse